eva_sub_cli_processing/sub_cli_to_eload_converter/json_to_xlsx_converter.py

```diff
@@ -153,90 +160,78 @@ def create_worksheet_header(self, workbook, worksheet_key, worksheet_title):
-                                                  value=json_to_xlsx_key_mapper[worksheet_key][header_key])
+                                                  value=header_value)
             cell.font = openpyxl.styles.Font(bold=True)
 
     def create_dummy_instructions_sheet(self, workbook):
         worksheet_title = 'PLEASE READ FIRST'
         workbook.create_sheet(worksheet_title)
         workbook[worksheet_title].cell(column=1, row=3, value='V1.1.4 August 2020')
 
-    def create_submitter_details_worksheet(self, workbook, worksheet_key, worksheet_title, submitter_details):
-        row_index = 1
-        for submitter in submitter_details:
+    def _fill_in_worksheet(self, output_workbook, output_worksheet_title, worksheet_key, source_data_list,  start_row_index=0):
+        row_index = start_row_index
+        for source_data_element in source_data_list:
             row_index += 1
             col_index = 0
-            for header_name in sorted(json_to_xlsx_key_mapper[worksheet_key]):
+            for header_key in sorted(json_to_xlsx_key_mapper[worksheet_key]):
                 col_index += 1
-                if header_name in submitter:
-                    workbook[worksheet_title].cell(column=col_index, row=row_index, value=submitter[header_name])
+                if header_key in source_data_element:
+                    cell_value = source_data_element.get(header_key)
+                    if isinstance(json_to_xlsx_key_mapper[worksheet_key][header_key], dict):
+                        header_dict = json_to_xlsx_key_mapper[worksheet_key][header_key]
+                        if 'transform' in header_dict and cell_value:
+                            cell_value = json_to_xlsx_key_mapper[worksheet_key][header_key].get('transform')(cell_value)
+
+                else:
+                    cell_value = ''
+                    if isinstance(json_to_xlsx_key_mapper[worksheet_key][header_key], dict):
+                        header_dict = json_to_xlsx_key_mapper[worksheet_key][header_key]
+                        if 'default' in header_dict and not cell_value:
+                            cell_value = json_to_xlsx_key_mapper[worksheet_key][header_key].get('default')
+                        if 'link' in header_dict and not cell_value:
+                            tmp_worksheet, tmp_header = json_to_xlsx_key_mapper[worksheet_key][header_key].get('link').split('.')
+                            cell_value = self.data[tmp_worksheet][tmp_header]
+                output_workbook[output_worksheet_title].cell(column=col_index, row=row_index, value=cell_value)
+
+    def create_submitter_details_worksheet(self, workbook, worksheet_key, worksheet_title, submitter_details):
+        row_index = 1
+        self._fill_in_worksheet(workbook, worksheet_title, worksheet_key, submitter_details, row_index)
 
     def create_project_worksheet(self, workbook, worksheet_key, worksheet_title, project_data):
```

Comment:
It's a bit odd to have separate methods for all of these now, maybe we can just call `fill_in_worksheet` with the appropriate row index directly?

Reply:
Though keeping `create_sample_worksheet` would be nice since there's still some logic there - up to you I guess.